fairmq/sdk/Topology.h

```diff
@@ -256,7 +256,21 @@ class BasicTopology : public AsioBase<Executor, Allocator>
     BasicTopology& operator=(const BasicTopology&) = delete;
 
     /// movable
-    BasicTopology(BasicTopology&&) = default;
+    BasicTopology(BasicTopology&& bt)
+        : AsioBase<Executor, Allocator>(bt.GetExecutor(), bt.GetAllocator())
+        , fDDSSession(std::move(bt.fDDSSession))
+        , fDDSTopo(std::move(bt.fDDSTopo))
+        , fStateData(std::move(bt.fStateData))
+        , fStateIndex(std::move(bt.fStateIndex))
+        , fMtx()
+        , fStateChangeUnsubscriptionCV()
+        , fHeartbeatsTimer(std::move(bt.fHeartbeatsTimer))
+        , fHeartbeatInterval(std::move(bt.fHeartbeatInterval))
+        , fChangeStateOps(std::move(bt.fChangeStateOps))
+        , fWaitForStateOps(std::move(bt.fWaitForStateOps))
+        , fSetPropertiesOps(std::move(bt.fSetPropertiesOps))
+        , fGetPropertiesOps(std::move(bt.fGetPropertiesOps))
+    {}
```

Comment:
Why?

If it is needed please create seperate commit with commit message containing the rationale.

Reply:
The default move ctor was implicitly deleted because mutex & condition_variable are not movable. Thus the implementation. I moved to a separate commit.

Reply:
I now packed both mtx and cv into unique_ptrs and reverted the move ctor back to ` = default`. Feel free to extend this if you want additional checks, or let me know.